src/links.tsx

```diff
@@ -92,17 +87,26 @@ export async function internalLinksTransform(
   opts: Options
 ): Promise<void> {
   const links = selectAll('link,linkBlock', tree) as Link[];
-  links.forEach(async link => {
-    if (!link || !link.url) return;
-    const resolver = opts.resolver;
-    const isLocal = resolver?.isLocal
-      ? resolver.isLocal(link.url)
-      : URLExt.isLocal(link.url);
-    if (isLocal) (link as any).internal = true;
-  });
+  await Promise.all(
+    links.map(async link => {
+      if (!link || !link.url) return;
+      const resolver = opts.resolver;
+      const href = link.url;
+      updateLinkTextIfEmpty(link, href);
+      const isLocal = resolver?.isLocal
+        ? resolver.isLocal(href)
+        : URLExt.isLocal(href);
+      if (!isLocal) return;
+      if (!resolver) return;
+      if ((link as any).static) {
+        // TODO: remove hash
+        const urlPath = await resolver.resolveUrl(href);
+        const url = await resolver.getDownloadUrl(urlPath);
+        (link as any).urlSource = href;
+        link.url = url;
```

Comment:
This is now an async function, and if the link is `static` we update the URL directly, otherwise we let Jupyter take care of it in the renderer.